### PR TITLE
Fix noexcept-ness in MakeArray

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

**Comment by mar-celine** [comment]
I will rebase #214 off of this PR

File: src/Utilities/MakeArray.hpp [hunk 1]
```diff
@@ -49,7 +55,10 @@ SPECTRE_ALWAYS_INLINE constexpr std::array<std::decay_t<T>, 0> make_array(
 /// helper function to make make_array<size_t> a simple call for users.
 template <std::size_t size, typename T>
 SPECTRE_ALWAYS_INLINE constexpr std::array<std::decay_t<T>, size> make_array(
-    std::integral_constant<std::size_t, size> /* unused */, T&& value) {
+    // clang-format off
```
Comment (kidder) [inline src/Utilities/MakeArray.hpp:58]: where is the matching clang-format on

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Need to test that the calculations work correctly

File: tests/Unit/Utilities/Test_MakeArray.cpp [hunk 2]
```diff
@@ -17,6 +17,30 @@ struct NonCopyable {
...
+ public:
+  DoesThrow() noexcept(false);
+  DoesThrow(const DoesThrow&) noexcept(false);
+  DoesThrow& operator=(const DoesThrow&) noexcept(false);
+  DoesThrow(DoesThrow&&) noexcept(false);
+  DoesThrow& operator=(DoesThrow&&) noexcept(false);
+};
+
+static_assert(noexcept(make_array<2>(0)),
+              "The assumed noexcept-ness is not what was calculated.");
```
Comment (nilsdeppe) [inline tests/Unit/Utilities/Test_MakeArray.cpp]: Failed Unit.Utilities.MakeArray testing noexcept calculation of make_array.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Squash the two commits into one, please.

File: tests/Unit/Utilities/Test_MakeArray.cpp [hunk 3]
```diff
@@ -17,6 +17,35 @@ struct NonCopyable {
...
+  ~DoesThrow() = default;
+};
+
+static_assert(noexcept(make_array<2>(0)),
+              "Failed Unit.Utilities.MakeArray testing noexcept calculation of "
+              "make_array.");
+static_assert(noexcept(make_array<2>(DoesNotThrow{})),
+              "Failed Unit.Utilities.MakeArray testing noexcept calculation of "
+              "make_array.");
+static_assert(!noexcept(make_array<2>(DoesThrow{})),
```
Comment (nilsdeppe) [inline tests/Unit/Utilities/Test_MakeArray.cpp]: please use `not`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**